### PR TITLE
Update facade handling.

### PR DESCRIPTION
**Pull request by frankban**
So that it is simpler to check for available facade
versions and to switch between endpoint implementations.

Initial work to drop support for Deployer GUI server facade.

### CODE REVIEW

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4430/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4431/
Test PASSed.

File: jujugui/static/gui/src/test/test_env_go.js [hunk 1]
```diff
@@ -329,19 +344,27 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
         // Assume login to be the first request.
         conn.msg({RequestId: 1, Response: {}});
         var environmentInfoMessage = conn.last_message(2);
+        console.log(environmentInfoMessage);
```
Comment (hatched) [inline jujugui/static/gui/src/test/test_env_go.js]: oops log left in.

File: jujugui/static/gui/src/test/test_env_go.js [hunk 2]
```diff
@@ -329,19 +344,27 @@ with this program.  If not, see <http://www.gnu.org/licenses/>.
...
           Request: 'EnvironmentInfo',
+          // Note that facade version here is 0 because the login mock response
+          // below is empty.
+          Version: 0,
           RequestId: 2,
           Params: {}
         };
         assert.deepEqual(environmentInfoExpected, environmentInfoMessage);
         var watchAllMessage = conn.last_message();
+        console.log(watchAllMessage);
```
Comment (hatched) [inline jujugui/static/gui/src/test/test_env_go.js]: log left in

**Comment by hatched** [comment]
:+1: Looks good!

File: jujugui/static/gui/src/test/test_env_go.js (hunk 1 above)
Comment (frankban) [inline jujugui/static/gui/src/test/test_env_go.js]: Done.

File: jujugui/static/gui/src/test/test_env_go.js (hunk 2 above)
Comment (frankban) [inline jujugui/static/gui/src/test/test_env_go.js]: Done.

**Comment by makyo** [comment]
:+1:

**Comment by jujugui** [comment]
Test FAILed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4434/
Test FAILed.

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/4435/
Test PASSed.

**Comment by frankban** [comment]
Thanks for the reviews!
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**